### PR TITLE
ref(api): Abstract Event Querystring Parsing Code

### PR DESCRIPTION
**Pull request by wedamija**

### CODE REVIEW

File: src/sentry/api/bases/organization.py [hunk 1]
```diff
@@ -91,6 +103,116 @@ class OrganizationDiscoverSavedQueryPermission(OrganizationPermission):
...
+                    organization=organization,
+                ).values_list('role', flat=True).get()
+            except OrganizationMember.DoesNotExist:
+                pass
+
+        if (
+            request.user.is_superuser
+            or (om_role and roles.get(om_role).is_global)
+            or organization.flags.allow_joinleave
+            or force_global_perms
```
Comment (wedamija) [inline src/sentry/api/bases/organization.py]: I added some extra permission things here based on what I saw on the release endpoint

File: src/sentry/api/bases/organization.py [hunk 2]
```diff
@@ -91,6 +103,116 @@ class OrganizationDiscoverSavedQueryPermission(OrganizationPermission):
 
 class OrganizationEndpoint(Endpoint):
     permission_classes = (OrganizationPermission, )
+    # Defines what happens if no date filter parameters are passed. If False,
+    # no date filtering occurs. If True, we provide default values.
+    DATE_FILTER_OPTIONAL = False
```
Comment (wedamija) [inline src/sentry/api/bases/organization.py]: Not totally sure this should be on the class, maybe could just parameterize `get_filter_params` more.

File: src/sentry/api/bases/organization.py [hunk 3]
```diff
@@ -91,6 +103,119 @@ class OrganizationDiscoverSavedQueryPermission(OrganizationPermission):
...
+                    user=request.user,
+                    organization=organization,
+                ).values_list('role', flat=True).get()
+            except OrganizationMember.DoesNotExist:
+                pass
+
+        if (
+            request.user.is_superuser
+            or (om_role and roles.get(om_role).is_global)
+            or organization.flags.allow_joinleave
```
Comment (macqueen) [inline src/sentry/api/bases/organization.py]: hmmm wondering if this should actually be here. i don't know that we've been super consistent about how this works in the product right now -- basically should a user see all projects they _could_ access or just the projects they currently have access to. maybe for now this should be another thing the endpoints can set themselves and then we can work on getting a decision on how the product should behave and we can figure out how to adjust accordingly.

File: src/sentry/api/bases/organization.py (hunk 2 above)
Comment (macqueen) [reply src/sentry/api/bases/organization.py]: yeah that might make it slightly less confusing, but either way

**Review by macqueen — APPROVED** [review]
ty for writing doc strings for all of these!

File: src/sentry/api/bases/organization.py (hunk 3 above)
Comment (wedamija) [reply src/sentry/api/bases/organization.py]: Yeah, I'm not sure what is correct here. You're probably right that it should be parameterized, will add that in.

File: src/sentry/api/bases/organization.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/api/bases/organization.py]: Yeah, I'll move this to params

**Pull request merged**